src/debugger/configuration/resolvers/attach.ts

```diff
@@ -50,33 +54,69 @@ export class AttachResolver implements vscode.DebugConfigurationProvider {
             }
         } else if (config.runtime === "Python") {
             const host = "localhost";
             const port = await port_finder.getPortPromise();
             const ptvsdInjectCommand = await utils.getPtvsdInjectCommand(host, port, config.processId);
 
-            const processOptions: child_process.ExecOptions = {
-                cwd: extension.baseDir,
-                env: extension.env,
+            const pythonattachdebugconfiguration: vscode.DebugConfiguration = {
+                name: `Python: ${config.processId}`,
+                type: "python",
+                request: "attach",
+                port: port,
+                host: host,
             };
-            child_process.exec(ptvsdInjectCommand, processOptions, (error, stdout, stderr) => {
-                if (!error) {
-                    const statusMsg = `New ptvsd instance running on ${host}:${port} injected into process [${config.processId}].`;
-                    extension.outputChannel.appendLine(statusMsg);
-                    extension.outputChannel.show(true);
-                    vscode.window.showInformationMessage(statusMsg);
-
-                    const pythonattachdebugconfiguration: vscode.DebugConfiguration = {
-                        name: `Python: ${config.processId}`,
-                        type: "python",
-                        request: "attach",
-                        port: port,
-                        host: host,
-                    };
-                    vscode.debug.startDebugging(undefined, pythonattachdebugconfiguration);
-                }
-            });
+
+            if (os.platform() === "win32") {
+                // ptvsd --pid works with child_process.exec() on Windows
+                const processOptions: child_process.ExecOptions = {
+                    cwd: extension.baseDir,
+                    env: extension.env,
+                };
+                child_process.exec(ptvsdInjectCommand, processOptions, (error, stdout, stderr) => {
+                    if (!error) {
+                        // @todo: add instructions for re-attaching to a Python process with ptvsd running
+                        const statusMsg = `New ptvsd instance running on ${host}:${port} injected into process [${config.processId}].`;
+                        extension.outputChannel.appendLine(statusMsg);
+                        extension.outputChannel.show(true);
+                        vscode.window.showInformationMessage(statusMsg);
+
+                        vscode.debug.startDebugging(undefined, pythonattachdebugconfiguration);
+                    }
+                });
+            } else {
+                // ptvsd --pid requires elevated permission on Ubuntu
+                const processOptions = {
+                    name: "ptvsd",
+                };
+                sudo.exec(ptvsdInjectCommand, processOptions, (error: string, stdout: string, stderr: string) => {
```

Comment:
Is it popping up a sudo dialog for password? I tried this in WSL with vscode-remote, and it doesn't do anything but just failed with errors. If you verify it worked in Ubuntu, I think it is fine for now but we probably need to revisit this piece soon.

Reply:
yes, it works on Ubuntu by popping up a UAC-like dialog